core/modules/memman/MemFile.h

```diff
@@ -0,0 +1,138 @@
+    //!
+    //! @return MLResult  When bLocked > 0 this number of bytes locked.
+    //!                   When bLocked = 0 no bytes were locked and retc holds
+    //!                   the reason as follows:
+    //! @return =0        retc = 0 the file did not meet minRefs restriction;
+    //!                   otherwise, it holds the errno failure code.
+    //-----------------------------------------------------------------------------
+
+    struct MLResult {size_t bLocked;
+                     int    retc;
+                     MLResult() {}
+                     MLResult(size_t lksz, int rc) : bLocked(lksz), retc(rc) {}
+                    };
+
+    MLResult    memLock(size_t maxBytes=0, int minRefs=0);
+
+    //-----------------------------------------------------------------------------
+    //! @brief Get number of active files (global count).
+    //!
+    //! @return The number of files.
+    //-----------------------------------------------------------------------------
+
+    static uint32_t numFiles();
+
+    //-----------------------------------------------------------------------------
+    //! @brief Obtain an object describing a in-memory file.
+    //!
+    //! @param  rc      - Reference to the place for an error code.
+    //! @param  fPath   - The path to the file.
+    //! @param  mem     - Reference to the memory object to use for the file.
+    //!
+    //! @return MFResult  When mfP is zero or retc is not zero, the MemFile
+    //!                   object could not be obtained and retc holds errno.
+    //-----------------------------------------------------------------------------
+
+    struct MFResult {MemFile* mfP;
+                     int      retc;
+                     MFResult() {}
+                     MFResult(MemFile* mfp, int rc) : mfP(mfp), retc(rc) {}
+                    };
+
+    static MFResult obtain(std::string const& fPath, Memory& mem);
+
+    //-----------------------------------------------------------------------------
+    //! @brief Release this table. Upon return it may not be references by
+    //!        the caller as it may have been deleted.
+    //-----------------------------------------------------------------------------
+
+    void release();
+
+private:
+    //-----------------------------------------------------------------------------
+    //! @brief Constructor. Only obtain() can allocate a MemFile object.
+    //!
+    //! @param  fPath   - The path to the file.
+    //! @param  mem     - Reference to the associated memory object.
+    //! @param  mInfo   - Initial value of the MemInfo object for the file.
+    //-----------------------------------------------------------------------------
+
+           MemFile(std::string const& fPath, Memory& mem, MemInfo const& minfo)
+                   : _fPath(fPath), _memory(mem), _memInfo(minfo),
+                     _refs(1), _isLocked(0) {}
+
+          ~MemFile() {}
```

Comment:
Constructor and destructor should be indented 4 spaces.

Default values for _refs and _isLocked could be defined with their declarations so they would only be defined in one place if multiple constructors are used.